src/Common/Infrastructure/RegistryApiController.cs

```diff
@@ -54,5 +57,23 @@ public abstract class RegistryApiController<T> : ApiController<T>
                 requestHeaders,
                 handleNotOkResponseAction,
                 cancellationToken);
+
+        protected string CreateCacheKeyForRequestQuery(string keyBaseValue)
+        {
+            if (string.IsNullOrWhiteSpace(keyBaseValue))
+                throw new ArgumentNullException(nameof(keyBaseValue));
+
+            bool ParameterHasValue(KeyValuePair<string, StringValues> parameter)
+                => !string.IsNullOrWhiteSpace(parameter.Value.ToString());
+
+            return Request.Query.Count == 0
+                ? keyBaseValue
+                : Request
+                    .Query
+                    .Where(ParameterHasValue)
+                    .OrderBy(queryParameter => queryParameter.Key)
+                    .Aggregate(keyBaseValue, (key, queryParameter) => $"{key}-{queryParameter.Key}:{queryParameter.Value}")
+                    .Replace(":-", ":");
```

Comment:
I think this needs to be removed, a queryparameter with a `-` in front means descending, so it is really a different cachekey

Reply:
replace ":-", ":" is only there to clean up keys like 
"legacy/parcel-list:-limit:100-offset:0" 
grouped parameter name+value for readability to "legacy/parcel-list:(limit:100)-(offset:0)"